### PR TITLE
Satellite5 deployment

### PR DESCRIPTION
**Pull request by rplevka**
Added a method `iso_install_5` to install sat5 from ISO - in order to be able to set up testing of the sat5->sat6 transition tool

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Please squash the commits in this PR.

    git checkout satellite5_deployment
    git rebase -i HEAD~5
    # review commit history to make sure it's sane
    git push -f origin satellite5_deployment

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -850,6 +850,80 @@ def iso_install(
...
+enable-tftp=yes
+EOF
+""".format(**opts))
+    # download a certificate
+    print('Downloading Certificate...')
+    run('wget {satellite-cert-file}'.format(**opts))
+    # ...and run the installer script.
+    with cd('/root/ISO'):
+        run('./install.pl --answer-file=/answers.txt --non-interactive \
+        --re-register --run-updater=yes --enable-tftp=yes')
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: run(
        './install.pl --answer-file=/answers.txt --non-interactive '
        '--re-register --run-updater=yes --enable=tftp=yes'
    )

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -850,6 +850,80 @@ def iso_install(
...
+rhn-http-proxy-password ={rhn-http-proxy-password}
+ssl-config-sslvhost =y
+ssl-set-org ={ssl-set-org}
+ssl-set-org-unit ={ssl-set-org-unit}
+ssl-set-city ={ssl-set-city}
+ssl-set-state ={ssl-set-state}
+ssl-set-country ={ssl-set-country}
+ssl-set-email={admin-email}
+ssl-password ={ssl-password}
+satellite-cert-file =/root/satellite-engineering-qa-5.7.cert
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: Is there significance to the whitespace before many of the "=" symbols in this file? It's different from what you'd write if placing these options directly on the command line.

**Comment by Ichimonji10** [comment]
ACK pending comments

**Comment by Ichimonji10** [comment]
> ./automation_tools/__init__.py:923:9: E128 continuation line under-indented for visual indent

**Comment by Ichimonji10** [comment]
flake8 is the only linting tool that *must* pass for a PR to be accepted. I also recomment reviewing code with pylint. For a non-trivial example of how to use pylint and flake8, see [this makefile](https://github.com/SatelliteQE/nailgun/blob/master/Makefile#L17-L23).

**Comment by Ichimonji10** [comment]
ACK

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -850,6 +850,80 @@ def iso_install(
         return installer_options
 
 
+def iso_install_5(
```
Comment (elyezer) [inline automation_tools/__init__.py]: Please name the task `satellite5_iso_insall`, now that we will be installing different product versions will be better to prefix the tasks with the product name + major version.

File: automation_tools/__init__.py [hunk 4]
```diff
@@ -850,6 +850,80 @@ def iso_install(
...
+    # Download the ISO
+    print('Downloading ISO...')
+    iso_download(iso_url)
+    # Create a 'check-out' folder, mount ISO to it...
+    run('mkdir -p ISO')
+    run('mount *.iso ISO -t iso9660 -o loop')
+
+    # prepare the answer file
+    opts = {
+        'admin-email': os.environ.get('ADMIN_EMAIL') or '',
```
Comment (elyezer) [inline automation_tools/__init__.py]: Better to use `os.environ.get('ADMIN_EMAIL', '')`. This applies for every line in this dictionary.

File: automation_tools/__init__.py [hunk 5]
```diff
@@ -850,6 +850,80 @@ def iso_install(
...
+        os.environ.get('RHN_HTTP_PROXY_PASSWORD') or '',
+        'ssl-set-org': os.environ.get('SSL_SET_ORG') or '',
+        'ssl-set-org-unit': os.environ.get('SSL_SET_ORG_UNIT') or '',
+        'ssl-set-city': os.environ.get('SSL_SET_CITY') or '',
+        'ssl-set-state': os.environ.get('SSL_SET_STATE') or '',
+        'ssl-set-country': os.environ.get('SSL_SET_COUNTRY') or '',
+        'ssl-password': os.environ.get('SSL_PASSWORD') or '',
+        'satellite-cert-file': os.environ.get('SATELLITE_CERT_FILE') or ''
+    }
+    run("""cat <<EOF > /answers.txt
```
Comment (elyezer) [inline automation_tools/__init__.py]: Unindented line is not so good when reading the code, you can have the same by changing to:

```python
run(
    'cat <<EOF > /answers.txt'
    'admin-email={admin-email}\n'
    'rhn-username={rhn-username}\n'
    'rhn-password={rhn-password}\n'
    'rhn-profile-name={rhn-profile-name}\n'
    'rhn-http-proxy={rhn-http-proxy}\n'
    'rhn-http-proxy-username={rhn-http-proxy-username}\n'
    'rhn-http-proxy-password={rhn-http-proxy-password}\n'
    'ssl-config-sslvhost=y\n'
    'ssl-set-org={ssl-set-org}\n'
    'ssl-set-org-unit={ssl-set-org-unit}\n'
    'ssl-set-city={ssl-set-city}\n'
    'ssl-set-state={ssl-set-state}\n'
    'ssl-set-country={ssl-set-country}\n'
    'ssl-set-email={admin-email}\n'
    'ssl-password={ssl-password}\n'
    'satellite-cert-file=/root/satellite-engineering-qa-5.7.cert\n'
    'enable-tftp=yes\n'
    'EOF\n'
)

File: automation_tools/__init__.py [hunk 6]
```diff
@@ -885,6 +959,7 @@ def product_install(distribution, create_vm=False, certificate_url=None,
         'satellite6-cdn': cdn_install,
         'satellite6-downstream': downstream_install,
         'satellite6-iso': iso_install,
+        'satellite5-iso': iso_install_5,
```
Comment (elyezer) [inline automation_tools/__init__.py]: When renamed, the task should be updated here.

**Comment by elyezer** [comment]
ACK, pending nitpick comments.

File: automation_tools/__init__.py (hunk 6 above)
Comment (rplevka) [inline automation_tools/__init__.py]: fixed

File: automation_tools/__init__.py (hunk 4 above)
Comment (Ichimonji10) [inline automation_tools/__init__.py]: :+1:

**Comment by Ichimonji10** [comment]
I think all of our comments have been addressed. ACK pending squash.

**Comment by Ichimonji10** [comment]
@elyezer, this looks good to me. Can you look over this PR and give it a sanity check? I want to make sure I haven't somehow messed it up.

**Comment by elyezer** [comment]
ACK

**Pull request merged**